tests/suites/test_suite_alignment.function

```diff
@@ -17,6 +17,18 @@ int parse_hex_string(char *hex_string, uint64_t *result)
     if (mbedtls_test_unhexify(raw, sizeof(raw), hex_string, &olen) != 0) {
         return 0;
     }
+
+    /* If < 8 bytes, shift right and pad with leading zeros for big-endian */
```

Comment:
This is all trying to compensate for the `if (MBEDTLS_IS_BIG_ENDIAN)` code below, but the `else` part of that is already portable — all you need to do is remove the big-endian stuff.

With https://github.com/gilles-peskine-arm/mbedtls/tree/pr_7440-1-fix the tests pass on platforms with either endianness with GCC (with `MBEDTLS_BSWAPxx is implemented using `__builtin_bswapxx`). I haven't tested other platforms (I don't have other cross-compilers readily available).